include/irr/asset/IMeshPacker.h

```diff
@@ -59,27 +62,39 @@ class IMeshPacker
 
 public:
 
-    IMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams, uint16_t maxIndexCountPerMDIData = std::numeric_limits<uint16_t>::max())
-        :m_maxIndexCountPerMDIData(maxIndexCountPerMDIData), /*TODO: delete when all alocators are done: */m_alctrReservedSpace{nullptr, nullptr, nullptr, nullptr}
+    IMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams, uint16_t minTriangleCountPerMDIData, uint16_t maxTriangleCountPerMDIData)
+        :m_maxTriangleCountPerMDIData(maxTriangleCountPerMDIData),
+         m_minTriangleCountPerMDIData(minTriangleCountPerMDIData),
+        m_MDIDataAlctrResSpc(nullptr), m_idxBuffAlctrResSpc(nullptr),
+        m_vtxBuffAlctrResSpc(nullptr), m_perInsVtxBuffAlctrResSpc(nullptr)
     {
         m_outVtxInputParams.enabledAttribFlags  = preDefinedLayout.enabledAttribFlags;
         m_outVtxInputParams.enabledBindingFlags = preDefinedLayout.enabledBindingFlags;
 
         memcpy(m_outVtxInputParams.attributes, preDefinedLayout.attributes, sizeof(m_outVtxInputParams.attributes));
 
-        //TODO: init allocators 
-        //m_alctrReservedSpace[0] = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator::reserved_size(/**/), _IRR_SIMD_ALIGNMENT);
-        //m_vtxBuffAlctr = core::GeneralpurposeAddressAllocator(m_alctrReservedSpace[0], 0u, 0u, 0u, allocParams.vertexBuffSupportedSizeInBytes / allocParams.vertexbufferMinAllocSize, allocParams.vertexbufferMinAllocSize);
+
+        m_vtxSize = calcVertexSize(preDefinedLayout, E_VERTEX_INPUT_RATE::EVIR_PER_VERTEX);
+        if (m_vtxSize)
+        {
+            m_vtxBuffAlctrResSpc = _IRR_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(std::max_align_t), allocParams.vertexBuffSupportedSizeInBytes / m_vtxSize, allocParams.vertexBufferMinAllocSize /*divided by vtxSize?*/), _IRR_SIMD_ALIGNMENT);
```

Comment:
if you don't count `allocParams.vertexBufferMinAllocSize` in bytes, but vertices then you don't need to divide, if you count in bytes, then you need to divide and the question is "do you round up?"

Reply:
likewise for `perInstanceVertexBufferMinAllocSize`